### PR TITLE
[Warmfix] Issue:146 Remove direct donation: support litecoin foundation

### PR DESCRIPTION
**Pull request by kcw-grunt**
## Problem
Previous binary was rejected by Apple as LF is not a charity (though is a Non-Profit).  They asked us to change the Donation working and functionality

## Issue
#146 

## Approach
The `Support Litecoin Foundation` button fires a sheet with and embedded webpage that the user can use to send grants to Litecoin Foundation
## Steps
- Removed Donation code
- Added more SwiftUI Code
- Added some Test Stubs
- Check the LTC address matches 
- Renamed `State` to `ReduxState` to avoid name collision

## Screenshot/GIF
|Resulting Action in Send|
|------------------------|
|SupportLF|

### CODE REVIEW

File: loafwallet/src/Views/SendViewCells/AddressCell.swift [hunk 1]
```diff
@@ -92,7 +92,7 @@ class AddressCell : UIView {
             scan.centerYAnchor.constraint(equalTo: centerYAnchor) ])
         paste.constrain([
             paste.centerYAnchor.constraint(equalTo: centerYAnchor),
-            paste.trailingAnchor.constraint(equalTo: scan.leadingAnchor, constant: -C.padding[1]) ])
+                            paste.trailingAnchor.constraint(equalTo: scan.leadingAnchor, constant: -5.0) ])
```
Comment (mosadialiou) [inline loafwallet/src/Views/SendViewCells/AddressCell.swift:95]: magic number here. Can't you use `C.padding`?

File: loafwallet/src/ViewControllers/RootModals/SendViewController.swift [hunk 2]
```diff
@@ -94,16 +94,17 @@ class SendViewController : UIViewController, Subscriber, ModalPresentable, Track
                 amountView.view.trailingAnchor.constraint(equalTo: view.trailingAnchor) ])
         })
         
-        donationCell.constrain([
-                  donationCell.widthAnchor.constraint(equalTo: amountView.view.widthAnchor),
-                  donationCell.topAnchor.constraint(equalTo: amountView.view.bottomAnchor),
-                  donationCell.leadingAnchor.constraint(equalTo: amountView.view.leadingAnchor),
-                  donationCell.heightAnchor.constraint(equalToConstant: 72.0)])
+        supportLitecoinFoundationCell.view.constrain([
```
Comment (mosadialiou) [inline loafwallet/src/ViewControllers/RootModals/SendViewController.swift:97]: is this well formatted? It's different from others

File: loafwallet/src/ViewControllers/RootModals/SendViewController.swift [hunk 3]
```diff
@@ -177,48 +176,18 @@ class SendViewController : UIViewController, Subscriber, ModalPresentable, Track
...
-                        self.sender.transaction = nil
-                    })
-                }
-                self.present(dynamicDonate, animated: true, completion: nil) 
-            }
+            
+            ///Paste in Support Litecoin Foundation address to textField
+        supportLitecoinFoundationCell.rootView.viewModel.didGetLTCAddress = { ltcAddress in
+            
+            ///Paste in Support Litecoin Foundation address to textField
```
Comment (mosadialiou) [inline loafwallet/src/ViewControllers/RootModals/SendViewController.swift:183]: don't we want to track this event?

File: loafwallet/src/Strings/zh-Hant.lproj/Localizable.strings [hunk 4]
```diff
@@ -1264,3 +1264,6 @@
 
 /* 1 of 3 */
 "WritePaperPhrase.step" = "%2$d 之 %1$d";
+
+/* Support the Litecoin Foundation */
+"SupportTheFoundation.title" = "";
```
Comment (mosadialiou) [inline loafwallet/src/Strings/zh-Hant.lproj/Localizable.strings:1269]: empty translation here

File: loafwallet/src/Strings/Base.lproj/Localizable.strings [hunk 5]
```diff
@@ -1263,3 +1263,9 @@
 
 /* Message for luxury fee */
 "FeeSelector.luxuryMessage" = "This option virtually guarantees acceptance of your transaction though you are paying a premium.";
+
+/* Support the Litecoin Foundation */
+"Support.title" = "";
+
+/* Support the Litecoin Foundation */
+"SupportTheFoundation.title" = "";
```
Comment (mosadialiou) [inline loafwallet/src/Strings/Base.lproj/Localizable.strings:1271]: use the same translation for english here

File: loafwallet/src/Strings/Base.lproj/Localizable.strings [hunk 6]
```diff
@@ -1263,3 +1263,9 @@
 
 /* Message for luxury fee */
 "FeeSelector.luxuryMessage" = "This option virtually guarantees acceptance of your transaction though you are paying a premium.";
+
+/* Support the Litecoin Foundation */
+"Support.title" = "";
```
Comment (mosadialiou) [inline loafwallet/src/Strings/Base.lproj/Localizable.strings:1268]: I think it's not used. If so to be removed

File: loafwallet/SupportLitecoinFoundationView.swift [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+    
+    var body: some View {
+        VStack {
+            Spacer()
+            Button(action: {
+                self.showSupportLFPage = true
+            }) {
+                
+                Text(S.SupportLitecoinFoundation.title)
+                    .padding(.all,10)
```
Comment (mosadialiou) [inline loafwallet/SupportLitecoinFoundationView.swift:42]: Not well formatted. there is a space after `,`
BTW format the whole class

**Review by mosadialiou — CHANGES_REQUESTED** [review]

File: loafwallet/src/SimpleRedux/ReduxState.swift [hunk 8]
```diff
@@ -8,7 +8,9 @@
 
 import UIKit
 
-struct State {
+
```
[the remark this answers is not in the document]
Comment (kcw-grunt) [reply loafwallet/src/SimpleRedux/ReduxState.swift:11]: This is the result of the Refactor.

File: loafwallet/src/Extensions/UIColor+Extension.swift [hunk 9]
```diff
@@ -156,3 +156,6 @@ extension UIColor {
         return UIColor(red: 0.0, green: 169.0/255.0, blue: 157.0/255.0, alpha: 1.0)
     }
 }
+
```
[the remark this answers is not in the document]
Comment (kcw-grunt) [reply loafwallet/src/Extensions/UIColor+Extension.swift:159]: This is the result of the Refactor.

File: loafwallet/src/Strings/ar.lproj/Localizable.strings [hunk 10]
```diff
@@ -1238,7 +1241,7 @@
 /* Corruption Error alert title */
 "Alert.corruptionMessage" = "";
 
-/* Donate Memo */
+/* Donate to the articles */
```
[the remark this answers is not in the document]
Comment (kcw-grunt) [reply loafwallet/src/Strings/ar.lproj/Localizable.strings:1244]: Actually all of the donation strings should be removed

File: loafwallet/src/Views/SendViewCells/AddressCell.swift (hunk 1 above)
Comment (kcw-grunt) [reply loafwallet/src/Views/SendViewCells/AddressCell.swift:95]: No...it only accepts an Int.  Added a double ability

File: loafwallet/src/ViewControllers/RootModals/SendViewController.swift (hunk 2 above)
Comment (kcw-grunt) [reply loafwallet/src/ViewControllers/RootModals/SendViewController.swift:97]: This is a SwiftUI view but it should be the same as the others....just fixed the indentation.

File: loafwallet/src/ViewControllers/RootModals/SendViewController.swift (hunk 3 above)
Comment (kcw-grunt) [reply loafwallet/src/ViewControllers/RootModals/SendViewController.swift:183]: Good point.

**Review by mosadialiou — APPROVED** [review]
LGTM

**Pull request merged**